language/move-core/types/src/vm_status.rs

```diff
@@ -42,22 +42,36 @@ pub static EXECUTION_STATUS_MIN_CODE: u64 = 4000;
 /// The maximum status code for runtim statuses
 pub static EXECUTION_STATUS_MAX_CODE: u64 = 4999;
 
-/// A `VMStatus` is represented as a required major status that is semantic coupled with with
-/// an optional sub status and message.
-#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Debug, Hash, Serialize, Deserialize)]
+/// A `VMStatus` is represented as either
+/// - `Executed` indicating successful execution
+/// - `Error` indicating an error from the VM itself
+/// - `MoveAbort` indicating an `abort` ocurred inside of a Move program
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
 #[cfg_attr(any(test, feature = "fuzzing"), derive(Arbitrary))]
 #[cfg_attr(any(test, feature = "fuzzing"), proptest(no_params))]
-pub struct VMStatus {
-    /// The major status, e.g. ABORTED, OUT_OF_GAS, etc.
-    pub major_status: StatusCode,
+pub enum VMStatus {
+    /// The VM status corresponding to an EXECUTED status code
+    Executed,
 
-    /// The optional sub status. Used e.g. for things such as the abort code, or the arithmetic
-    /// error type for an ARITHMETIC_ERROR major status.
-    pub sub_status: Option<u64>,
+    /// Indicates an error from the VM, e.g. OUT_OF_GAS, INVALID_AUTH_KEY, RET_TYPE_MISMATCH_ERROR
+    /// etc.
+    /// The code will neither EXECUTED nor ABORTED
+    Error(StatusCode),
```

Comment:
so changing `StatusCode` to a more generic one we'll think about, correct?
To be honest I'd be even fine with `Error` and nothing else, like `Executed`

Reply:
Sadly collapsing to just `Error` won't work; the status code is used to determine whether the transaction should be included in the transaction accumulator as well.

Reply:
Could we maybe condense the amount of codes though? Basically: All Status Codes relating to "libra errors" + a few (<10) codes for "VM errors"

Reply:
yeah I'd like that, I'd like the strictly needed for the chain to function